### PR TITLE
Ft cpdt 215

### PR DESCRIPTION
**Pull request by fobesjason**
## Type of change

- [ ] Bug Fix
- [x] New Feature
- [ ] Refactor
- [ ] Project Build
- [ ] Project Configuration

## Submission Checklist

- [x] Have you followed the guidelines in our [Contributing document](https://github.com/fanappics/buoy/blob/master/CONTRIBUTING.md)?
- [x] Did you create appropriate tests?

## Description

New popup component for information and confirmation purposes

## Related Issues / Bugs / Feature Requests

Link to any appropriate GitHub/Jira issues, bugs or freature requests

## Test Deployment URL

Replace branch-name with the name of your branch.

https://rawgit.com/fanappics/buoy/ft-cpdt-215/docs/index.html

## Deployment Notes

Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.

### CODE REVIEW

**Comment by kierachell** [comment]
- seems like elements under the popup are clickable. Expected?
- should the modal capture focus when it shows up? e.g when there are buttons inside the modal
click_under

File: src/components/BButton/BButton.vue [hunk 1]
```diff
@@ -4,6 +4,7 @@
     :class="computedClass"
     :disabled="disabled"
     :style="computedStyle"
+    @click="clickHandler($event)"
```
Comment (mlear) [inline src/components/BButton/BButton.vue]: We decided on event handler styling, i think it was `onEventName`?

**Comment by fobesjason** [comment]
@kierachell It was a Firefox specific problem only with the docs.  I fixed it

**Comment by kierachell** [comment]
Works now!

Maybe a question to @eleanormal : 
- should the modal autofocus after click?
- should the modal trap focus (you can't tab out of it until it's closed)?

**Comment by eleanormal** [comment]
On the confirmation example, the text next to the button moves when the popup opens - that shouldn't happen

**Comment by eleanormal** [comment]
Popup doesn't need autofocus - that could be an optional feature we add later as a prop.
Modal should not trap focus.
When focus leaves the modal, the modal should close.
When "Confirm" or "Cancel" is it, focus should go back to the button that triggered it (unless that confirm/cancel action makes something else happen like page navigation or refresh or something)

**Pull request merged**